tools/nixpkgs-hammer

```diff
@@ -23,34 +99,44 @@ def get_ast_check_programs():
     return set()
 
 
-def ast_checks(
-    attr_positions: Dict[str, str], excluded_rules: List[str]
-) -> Dict[str, List]:
-    file_to_attr = defaultdict(list)
-    for attr, file_colon_line in attr_positions.items():
-        file = file_colon_line.rsplit(":", 1)[0]
-        if file:
-            file_to_attr[file].append(attr)
+def run_external_checks(
+    attrs: List[Attr], excluded_rules: List[str]
+) -> Dict[str, List[Report]]:
+    rules = get_check_programs() - set(excluded_rules)
+    if len(rules) == 0:
+        return {}
 
-    rules = get_ast_check_programs() - set(excluded_rules)
+    names_without_outputs = [a for a in attrs if a.output is None or not a.output.exists()]
+    if len(names_without_outputs) > 0:
+        quoted_name = ", ".join([f"‘{a.name}’" for a in names_without_outputs])
+        print(red(f"""Warning: packages have not yet been built: {quoted_name}.
```

Comment:
Could you use dedent here instead of the unbalanced indentation?

Also, rather than printing, return a `Report`.